.github/workflows/e2e-tests-atomic.yml

```diff
@@ -0,0 +1,70 @@
+name: "E2E Tests on Atomic - All"
+
+# This action will run when it is triggered manually or on Pull Request with base=trunk
+on:
+  pull_request:
+    branches:
+      - trunk
+  workflow_dispatch:
+
+env:
+  WC_E2E_SCREENSHOTS: 1
+  E2E_SLACK_CHANNEL: ${{ secrets.E2E_SLACK_CHANNEL }}
+  E2E_SLACK_TOKEN: ${{ secrets.E2E_SLACK_TOKEN }}
+  E2E_WP_VERSION: 'nightly'
+  E2E_WC_VERSION: 'latest'
+  E2E_RESULT_FILEPATH: 'tests/e2e/results.json'
+  NODE_ENV: 'atomic'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Run tests against WP Nightly & WC latest. Not UPE ones to avoid impacting the other tests with UPE enabled.
+  wp-nightly-tests-first-batch:
+    name: "WP - nightly | WC - latest | ${{ matrix.test_groups }} - ${{ matrix.test_branches }}"
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        test_groups: [ 'wcpay', 'subscriptions' ]
+        test_branches: [ 'merchant', 'shopper' ]
+        exclude:
+          - test_groups: 'subscriptions'
+    
+    env:
+      E2E_GROUP: ${{ matrix.test_groups }}
+      E2E_BRANCH: ${{ matrix.test_branches }}
+    
+    
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+
+      - name: "Run the tests"
+        uses: ./.github/actions/e2e/atomic-prepare-and-run
+  
+  # Run UPE tests against WP Nightly & WC latest.
+  wp-nightly-tests-upe-batches:
+    name: "WP - nightly | WC - latest | ${{ matrix.test_groups }} - ${{ matrix.test_branches }}"
+    runs-on: ubuntu-22.04
+    needs: wp-nightly-tests-first-batch
+    if: success() || failure()
+    strategy:
+      fail-fast: false
+      max-parallel: 1 # Only one UPE group at a time to avoid conflicts between them
```

Comment:
The `upe` and `upeSplit` can't be executed at the same time (conflicts on the setting itself), but to keep the file concise, I gathered them in a same matrix